File: src/main/java/org/flossware/civilization/model/SimulationRules.java
package org.flossware.civilization.model;

/**
 * Simulation execution rules.
 */
public record SimulationRules(
    String timeStep,
    boolean deterministicReproducible,
    long baseRandomSeed,
    int monteCarloRuns,
    int parallelThreads
) {
    public SimulationRules {
        if (monteCarloRuns < 1) {
            throw new IllegalArgumentException("Monte Carlo runs must be at least 1");
        }
        if (parallelThreads < 1) {
            throw new IllegalArgumentException("Parallel threads must be at least 1");
        }
    }
}

File: mobile/shared/src/commonMain/kotlin/org/flossware/civilization/model/SimulationRules.kt
package org.flossware.civilization.model

import kotlinx.serialization.Serializable

@Serializable
data class SimulationRules(
    val timeStep: String,
    val deterministicReproducible: Boolean,
    val baseRandomSeed: Long,
    val monteCarloRuns: Int,
    val parallelThreads: Int
) {
    init {
        require(monteCarloRuns >= 1) { "Monte Carlo runs must be at least 1" }
        require(parallelThreads >= 1) { "Parallel threads must be at least 1" }
    }
}
